node/runtime/src/lib.rs

```diff
@@ -985,136 +988,126 @@ impl Runtime {
             }
-            Transaction::Receipt(ref r) => {
-                if account_to_shard_id(&r.receiver) == shard_id {
-                    let mut tmp_new_receipts = vec![];
-                    let apply_result = runtime.apply_receipt(state_update, r, &mut tmp_new_receipts, block_index, &mut result.logs);
-                    for receipt in tmp_new_receipts.iter() {
-                        if let Transaction::Receipt(r) = receipt {
-                            result.receipts.push(r.nonce);
-                        }
-                    }
-                    match apply_result {
-                        Ok(()) => {
-                            state_update.commit();
-                            new_receipts.append(&mut tmp_new_receipts);
-                            result.status = TransactionStatus::Completed;
-                        }
-                        Err(s) => {
-                            debug!(target: "runtime", "{}", s);
-                            state_update.rollback();
-                            new_receipts.append(&mut tmp_new_receipts);
-                            result.status = TransactionStatus::Failed;
-                        }
-                    };
-                } else {
-                    // wrong receipt
-                    debug!(target: "runtime", "receipt sent to the wrong shard");
-                    result.status = TransactionStatus::Failed;
-                }
+            Err(s) => {
+                debug!(target: "runtime", "{}", s);
+                state_update.rollback();
+                result.status = TransactionStatus::Failed;
             }
         };
         result
     }
 
-    /// check whether transactions in a block are valid and return the new root
-    /// if they are
-    pub fn check(
-        &mut self,
-        apply_state: &ApplyState,
-        prev_receipts: &[Transaction],
-        transactions: &[Transaction],
-    ) -> Option<(storage::DBChanges, MerkleHash, Vec<TransactionResult>)> {
-        let mut new_receipts = vec![];
-        let mut state_update = StateDbUpdate::new(self.state_db.clone(), apply_state.root);
-        let mut authority_proposals = vec![];
-        let shard_id = apply_state.shard_id;
-        let block_index = apply_state.block_index;
-        let mut tx_result = vec![];
-        for tx in prev_receipts.iter().chain(transactions) {
-            tx_result.push(Self::process_transaction(
-                self,
-                &mut state_update,
-                shard_id,
+    fn process_receipt(
+        runtime: &mut Self,
+        state_update: &mut StateDbUpdate,
+        shard_id: ShardId,
+        block_index: BlockIndex,
+        receipt: &ReceiptTransaction,
+        new_receipts: &mut HashMap<ShardId, Vec<ReceiptTransaction>>,
+    ) -> TransactionResult {
+        let mut result = TransactionResult::default();
+        if account_to_shard_id(&receipt.receiver) == shard_id {
```

Comment:
```
if account_to_shard_id(&receipt.receiver) != shard_id {
    debug!(target: "runtime", "receipt sent to the wrong shard");
    result.status = TransactionStatus::Failed;
  return result;
}
```
also let's instead of debug! -> put things into logs.
And we should add printing logs at the end of transaction processing.

Reply:
Fixed in 64bf1a3